raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/ImplicitUriParametersInjectionTransformer.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2013 (c) MuleSoft, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+package org.raml.v2.internal.impl.v10.phase;
+
+import org.raml.v2.internal.impl.commons.nodes.ResourceNode;
+import org.raml.v2.internal.impl.v10.nodes.NativeTypeExpressionNode;
+import org.raml.v2.internal.impl.v10.nodes.PropertyNode;
+import org.raml.yagi.framework.nodes.KeyValueNode;
+import org.raml.yagi.framework.nodes.KeyValueNodeImpl;
+import org.raml.yagi.framework.nodes.Node;
+import org.raml.yagi.framework.nodes.ObjectNodeImpl;
+import org.raml.yagi.framework.nodes.SimpleTypeNode;
+import org.raml.yagi.framework.nodes.StringNodeImpl;
+import org.raml.yagi.framework.phase.Transformer;
+import org.raml.yagi.framework.util.NodeSelector;
+
+import java.util.List;
+
+import static org.raml.v2.internal.impl.commons.phase.UnusedParametersTransformer.getUriTemplates;
+
+public class ImplicitUriParametersInjectionTransformer implements Transformer
+{
+    final private static String URI_PARAMETERS = "uriParameters";
+
+    @Override
+    public boolean matches(Node node)
+    {
+        return node instanceof ResourceNode;
+    }
+
+    @Override
+    public Node transform(Node node)
+    {
+        List<String> templates = getUriTemplates(((ResourceNode) node).getRelativeUri());
```

Comment:
`getUriTemplates` is a method of another transformer. Maybe a better approach would be moving it to a util class. What do you think?

Reply:
done